tests/test_date.py

```diff
@@ -239,59 +406,163 @@ def test_should_return_yesterday(self, date_string):
-                         date_data['date_obj'].date(), date_string)
-
-    def test_should_not_assume_language_too_early(self):
-        date_fixtures = [
-            (u'07/07/2014', datetime(2014, 7, 7)),
-            (u'07.jul.2014 | 12:52', datetime(2014, 7, 7)),
-            (u'07.ago.2014 | 12:52', datetime(2014, 8, 7)),
-            (u'07.feb.2014 | 12:52', datetime(2014, 2, 7)),
-            (u'07.ene.2014 | 12:52', datetime(2014, 1, 7)),
-        ]
-
-        for date_string, correct_date in date_fixtures:
-            date_data = self.parser.get_date_data(date_string)
-            self.assertIsNotNone(date_data['date_obj'],
-                                 "Could not get date for: %s" % date_string)
-            self.check_equal(correct_date.date(),
-                             date_data['date_obj'].date(), date_string)
+        self.given_date_string(date_string)
+        self.given_today(datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0))
+        self.given_day_before_yesterday()
+        self.when_date_data()
+        self.then_should_return_day_before_yesterday_is_not_none()
+        self.then_should_return_day_before_yesterday_is_equal()
+
+    @parameterized.expand([
+        param(u'07/07/2014', datetime(2014, 7, 7)),
+        param(u'07.jul.2014 | 12:52', datetime(2014, 7, 7)),
+        param(u'07.ago.2014 | 12:52', datetime(2014, 8, 7)),
+        param(u'07.feb.2014 | 12:52', datetime(2014, 2, 7)),
+        param(u'07.ene.2014 | 12:52', datetime(2014, 1, 7)),
+    ])
+    def test_should_not_assume_language_too_early(self, date_string, correct_date):
+        self.given_date_string(date_string)
+        self.given_correct_date(correct_date)
+        self.when_date_data()
+        self.then_should_not_assume_language_too_early_is_not_none()
+        self.then_should_not_assume_language_too_early_is_equal()
 
     def test_should_enable_redetection_for_multiple_languages(self):
-        date_fixtures = [
-            (u'13 Ago, 2014', datetime(2014, 8, 13)),
-            (u'11 Marzo, 2014', datetime(2014, 3, 11)),
-            (u'13 Septiembre, 2014', datetime(2014, 9, 13)),
-            (u'13 Setembro, 2014', datetime(2014, 9, 13)),
-            (u'13 Março, 2014', datetime(2014, 3, 13)),
-        ]
-        parser = date.DateDataParser(allow_redetect_language=True)
-
-        for date_string, correct_date in date_fixtures:
-            date_data = parser.get_date_data(date_string)
-            self.assertEqual(correct_date.date(), date_data['date_obj'].date())
+        dates_to_parse = {u'13 Ago, 2014': datetime(2014, 8, 13),
+                          u'11 Marzo, 2014': datetime(2014, 3, 11),
+                          u'13 Septiembre, 2014': datetime(2014, 9, 13),
+                          u'13 Setembro, 2014': datetime(2014, 9, 13),
+                          u'13 Março, 2014': datetime(2014, 3, 13)}
+
+        self.given_languages_parser()
+        self.given_known_languages('en', 'fr', 'cz', 'ru')
```

Comment:
You used languages from my example, while you should use languages in which dates from `dates_to_parse` are written.